config/checkstyle_checks.xml

```diff
@@ -425,6 +425,13 @@
       <message key="matchxpath.match"
                value="Avoid using deprecated method 'DetailAst.branchContains()'."/>
     </module>
+    <module name="MatchXpath">
+      <property name="id" value="commentStartWithSpace"/>
+      <property name="query"
+                value="//SINGLE_LINE_COMMENT[./COMMENT_CONTENT[not(starts-with(@text, ' '))
+                       and not(@text = '\n') and not(ends-with(@text, '//\n')) ]]"/>
```

Comment:
Can we do the same multi-line comments ?
I think ToDoComment check support both.

Please share output of your testing that it works.

Reply:
> I think ToDoComment check support both
 
Looking at regexp, I have doubts if it supported both. `[^\s\/*]` regexp would match almost all block comments due to the inclusion of `\n`.

Anyways, I have included support for block comments. Javadoc comments need not be checked by the `MatchXpath` check as it has its own `JavadocBlockLocation` and `JavadocContentLocation` configs.